File: de_php_a_python/src/list_comp.php
<?php

$frutas = ["manzana", "sandia", "pomelo", "mandarina"];
$empiezan_con_m = [];

foreach ($frutas as $fruta) {
    if (str_starts_with($fruta, "m")) {
        $empiezan_con_m[] = $fruta;
    }
}
var_dump($empiezan_con_m);

// Intentando hacer listas por comprensión en PHP
$frutas = ["manzana", "sandia", "pomelo", "mandarina"];
$empiezan_con_m = array_filter($frutas, function($fruta){return str_starts_with($fruta, "m");});
var_dump($empiezan_con_m);

File: de_php_a_python/src/list_comp.py
frutas = ["manzana", "sandia", "pomelo", "mandarina"]
empiezan_con_m = []

for fruta in frutas:
    if fruta.startswith("m"):
        empiezan_con_m.append(fruta)

print(empiezan_con_m)


# Ahora que somos pythonistas...

frutas = ["manzana", "sandia", "pomelo", "mandarina"]
empiezan_con_m = [fruta for fruta in frutas if fruta.startswith("m")]
print(empiezan_con_m)
